packages/server/src/MessageProcessor.js

```diff
@@ -362,9 +361,9 @@ export class MessageProcessor {
   async handleCompletionRequest(
     params: CompletionRequest.type,
     token: CancellationToken,
-  ): Promise<CompletionList | Array<CompletionItem>> {
+  ): Promise<?CompletionList> {
     if (!this._isInitialized) {
-      return [];
+      return null;
```

Comment:
Seems like these are breaking changes; previously no `null` was ever returned. MessageProcessor is exported in `index.js`, so others _could_ be relying on it? I don't currently have enough context to know if this is a major issue or not.

Reply:
agreed, I can look into it more, it looks like it could have been a legacy code issue, because there was a second case for `Array<CompletionItem>` that is no longer being used. I'll look deeper down the callstack and see what impact this has. I guess I could change the function declaration to return `CompletionList | Array<any>` to keep this in place?

Reply:
What’s the Flow error you’re addressing with this? I’d prefer `$FlowFixMe` on the return atatement(s) over `any` in this instance, I think.

Reply:
Seconded, it would be good to know what error is being fixed.

If this method is intended to conform to [CompletionItemProvider](https://code.visualstudio.com/api/references/vscode-api#CompletionItemProvider) (I am not sure if this is the case) then both `Array<CompletionItem>` and `null` are valid. If you're just looking to clean up the types, then it may be possible to create an empty `CompletionList` for these cases.

Reply:
When upgrading to the latest flow, returning an empty array is considered invalid, I'll pull up the specific error here shortly

Reply:
Here's the error against flow `0.100.0`
> Could not decide which case to select. Since case 1 [1] may work but if it doesn't case 2 [2] looks promising too. To fix add a type annotation to unknown element type of empty array [3].

it expects `Promise<CompletionList | Array<CompletionItem>` as a response, so returning an empty array seems to clash with that IMO. some places I've read say that in flow, an empty array should be  typed as`Array<any>` which also doesn't look very good.

Reply:
I think you can do:
```js
const emptyResult: Array<CompletionItem> = [];
return emptyResult;
```

or:

```js
return ([]: Array<CompletionItem>);
```

(I prefer the former.)